### DISCUSSION TITLE
Copilot cannot offer suggestions in between words / code

### DISCUSSION THREAD

**Discussion by Maxiviper117** [community/community#69002] [Copilot Conversations]
### Select Topic Area

Product Feedback

## Copilot Suggestion Limitations

Currently, Copilot seems to have specific behavior when offering suggestions based on the cursor's position. This behavior seems more restrictive in certain scenarios.

### Current Behavior
1. **Suggestions In-Between Words**: Copilot does not provide suggestions when the cursor is placed in the middle of a line or a statement.
   
   - Example: ![image](https://github.com/community/community/assets/11472492/d4836593-8e47-46e1-9d42-b5f0c46bbb11)
   
2. **Suggestions at End of Line**: Copilot does offer suggestions when the cursor is at the end of a line.

   - Example: ![image](https://github.com/community/community/assets/11472492/f7acbfa0-1f69-4f76-b497-faef7d92e3cb)

3. **In-Code Limitations**: The same restriction applies to code. However, exceptions occur when the cursor is placed between specific code characters.

   - No suggestion when cursor is not at end: ![image](https://github.com/community/community/assets/11472492/f2025fc1-fc75-4bc6-890c-06bec547f54a)
   - Suggestion between characters such as the `(...)`: ![image](https://github.com/community/community/assets/11472492/3a2c3b95-777d-48c3-b701-6e9004ce5dad)

### Comparison with Codeium

In contrast, another AI code completion tool named Codeium seems to handle these scenarios more flexibly.

1. **In-Between Word Suggestions**: Codeium can provide suggestions even when the cursor is placed in the middle of words.

   - Example: ![image](https://github.com/community/community/assets/11472492/2ce5f876-d04a-4084-8f7a-4d1334b0dfa3)

2. **Code Suggestion Capabilities**: It also offers more intuitive suggestions in places where Copilot appears to be limited.

   - Example: ![image](https://github.com/community/community/assets/11472492/6ac7097b-86fe-4335-a515-04c60c345372)

### Would be nice if Copilot could do this.

**Comment by JuanCabre** [comment]
I am still very interested in this feature. Is there any update?

**Reply by otomutachi** [reply]
Not all

**Comment by naktinis** [comment]
Seems to be a pretty common feature nowadays in other AI autocompletion tools like Cursor. Any chance this could get added?

**Reply by otomutachi** [reply]
Nop

**Comment by Maxiviper117** [comment]
Still issue till this day, and the Next Edit Suggestion feature is wayyy too unreliable to pick up these intricacies.

**Comment by mirao** [comment]
Here I entered "`so that`" and no suggestion is offered by Copilot. IMHO Copilot should be able to suggest in such case because it's clear that it's at the end of the string.
![image](https://github.com/user-attachments/assets/9acbf4f0-0009-41fc-8f02-a232ba3cfe56)

a workaround is removing of the rest of line. Then suggestion ("`it can be taken today`") is offered well
![image](https://github.com/user-attachments/assets/1e5e2cad-eea3-45f6-8f08-7deed933923d)

**Reply by mirao** [reply]
> Here I entered "`so that`" and no suggestion is offered by Copilot. IMHO Copilot should be able to suggest in such case because it's clear that it's at the end of the string.

It seems that by some updates now Copilot is able to suggest in the middle of a string.
I'm just testing the issue I had two weeks ago and it works as expected. Tested with VS Code 1.102.2

**Comment by Tikolu** [comment]
I cannot believe that it has been two years and this simple feature is still missing.

**Comment by SRBlinds** [comment]
Without mid-line suggestions, it slows down the workflow and limits the usefulness of Copilot. Fixing this would really help make Copilot more accessible and effective for SR Blinds users.

**Reply by mirao** [reply]
It already works for me, see my comment https://github.com/orgs/community/discussions/69002#discussioncomment-13888042

**Reply by Tikolu** [reply]
> It already works for me, see my comment https://github.com/orgs/community/discussions/69002#discussioncomment-13888042

Still does not work for me
